### PR TITLE
Send Triggered Job to App

### PR DESCRIPTION
**Pull request by cicoyle**
Successfully sending triggered jobs back to the app via the app channel from the daprd sidecar using both grpc and http protocols. Tests added for both protocols :)

See my [test-apps repo](https://github.com/cicoyle/test-apps) I've been using for testing where I have 2 apps, one for grpc comms and one for http where both are able to receive the triggered job.

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/integration/suite/daprd/scheduler/streaming/stream.go [hunk 1]
```diff
@@ -96,6 +97,7 @@ func (s *streaming) Setup(t *testing.T) []framework.Option {
 		daprd.WithExecOptions(
 			exec.WithStdout(s.streamloglineDaprB.Stdout()),
 		),
+		daprd.WithLogLevel("debug"),
```
Comment (cicoyle) [inline tests/integration/suite/daprd/scheduler/streaming/stream.go:100]: I thought this test was still valuable, as it ensures that the sidecar receives the job

File: pkg/channel/grpc/grpc_channel.go [hunk 2]
```diff
@@ -86,6 +87,69 @@ func (g *Channel) InvokeMethod(ctx context.Context, req *invokev1.InvokeMethodRe
...
+
+func (g *Channel) sendJob(ctx context.Context, req *invokev1.InvokeMethodRequest) (*invokev1.InvokeMethodResponse, error) {
+	if g.ch != nil {
+		g.ch <- struct{}{}
+	}
+
+	// Read the request, including the data
+	pd, err := req.ProtoWithData()
+	if err != nil {
+		return nil, err
```
Comment (JoshVanL) [inline pkg/channel/grpc/grpc_channel.go:107]: Defer releasing ch

File: pkg/channel/grpc/grpc_channel.go [hunk 3]
```diff
@@ -115,7 +179,7 @@ func (g *Channel) invokeMethodV1(ctx context.Context, req *invokev1.InvokeMethod
 		grpc.MaxCallSendMsgSize(g.maxRequestBodySize),
 		grpc.MaxCallRecvMsgSize(g.maxRequestBodySize),
 	}
-
+	//client.OnTriggerJobEvent() in triggerJob
```
Comment (JoshVanL) [inline pkg/channel/grpc/grpc_channel.go]: Remove

File: pkg/channel/http/http_channel.go [hunk 4]
```diff
@@ -170,6 +170,135 @@ func (h *Channel) InvokeMethod(ctx context.Context, req *invokev1.InvokeMethodRe
...
+	verb := msg.GetHttpExtension().GetVerb().String()
+	method := msg.GetMethod()
+
+	// Construct the URL using the base address from the channel
+	uri := strings.Builder{}
+	uri.WriteString(h.baseAddress)
+
+	if len(method) > 0 && method[0] != '/' {
+		uri.WriteRune('/')
+	}
```
Comment (JoshVanL) [inline pkg/channel/http/http_channel.go:191]: What does this do?

**Review by JoshVanL — CHANGES_REQUESTED** [review]

File: pkg/channel/http/http_channel.go (hunk 4 above)
Comment (cicoyle) [reply pkg/channel/http/http_channel.go:191]: It's a check to ensure that the URL starts with a forward slash. I kept this check for consistency with the service invocation logic, but can rm if you feel strongly that it should be removed.

File: pkg/channel/grpc/grpc_channel.go (hunk 2 above)
Comment (cicoyle) [reply pkg/channel/grpc/grpc_channel.go:107]: done

**Review by JoshVanL — APPROVED** [review]

File: pkg/channel/http/http_channel.go (hunk 4 above)
Comment (cicoyle) [reply pkg/channel/http/http_channel.go:191]: I tried removing this and it fails the `httpcallback/run/app_receives_triggered_job` so I believe this is necessary at least for now. I'm going to merge as is.

**Pull request merged**